src/overlays/actors/ovl_En_Light/z_en_light.c

```diff
@@ -28,16 +30,247 @@ const ActorInit En_Light_InitVars = {
+        75,
+    },
+    {
+        { 255, 170, 255, 255 },
+        { 100, 0, 255 },
+        75,
+    },
+    {
+        { 170, 255, 255, 255 },
+        { 0, 0, 255 },
+        75,
+    },
+    {
+        { 170, 255, 255, 255 },
+        { 0, 150, 255 },
+        75,
+    },
+};
+
+void EnLight_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnLight* this = THIS;
+
+    if (!ENLIGHT_GET_4000(&this->actor)) {
+        if ((gSaveContext.gameMode == 3) || ENLIGHT_GET_2000(&this->actor)) {
+            Lights_PointNoGlowSetInfo(&this->lightInfo, this->actor.world.pos.x,
+                                      ((this->actor.params < 0) ? 1 : 40) + (s32)this->actor.world.pos.y,
+                                      this->actor.world.pos.z, 255, 255, 180, -1);
+        } else {
+            Lights_PointGlowSetInfo(&this->lightInfo, this->actor.world.pos.x,
+                                    ((this->actor.params < 0) ? 1 : 40) + (s32)this->actor.world.pos.y,
+                                    this->actor.world.pos.z, 255, 255, 180, -1);
+        }
+        this->lightNode = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfo);
+    }
+
+    Actor_SetScale(&this->actor, D_808666D0[ENLIGHT_GET_F(&this->actor)].unk_07 * 0.0001f);
+
+    this->unk_144 = (s8)(Rand_ZeroOne() * 255.0f);
+
+    if (ENLIGHT_GET_800(&this->actor)) {
+        this->actor.update = func_80865F38;
+        if (ENLIGHT_GET_1000(&this->actor) && Flags_GetSwitch(globalCtx, ENLIGHT_SWITCHFLAG(&this->actor))) {
+            Actor_SetScale(&this->actor, 0.0f);
+        }
+    } else if ENLIGHT_GET_2000 (&this->actor) {
+        this->actor.draw = NULL;
+    }
+}
+
+void EnLight_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnLight* this = THIS;
+
+    if (!ENLIGHT_GET_4000(&this->actor)) {
+        LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+    }
+}
+
+void func_80865BF8(EnLight* this, GlobalContext* globalCtx) {
+    this->actor.shape.rot.y = BINANG_ROT180(Camera_GetCamDirYaw(GET_ACTIVE_CAM(globalCtx)));
+
+    if (this->actor.parent != NULL) {
+        Math_Vec3f_Copy(&this->actor.world.pos, &this->actor.parent->world.pos);
+        this->actor.world.pos.y += 17.0f;
+    }
+
+    this->unk_144++;
+}
+
+void EnLight_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnLight* this = THIS;
+
+    if (!ENLIGHT_GET_4000(&this->actor)) {
+        EnLightStruct* sp28 = &D_808666D0[ENLIGHT_GET_F(&this->actor)];
+        f32 temp_f2 = (Rand_ZeroOne() * 0.5f) + 0.5f;
+
+        Lights_PointSetColorAndRadius(&this->lightInfo, (u8)(sp28->unk_00.r * temp_f2), (u8)(sp28->unk_00.g * temp_f2),
+                                      (u8)(sp28->unk_00.b * temp_f2),
+                                      (this->actor.params < 0)         ? 100
+                                      : ENLIGHT_GET_2000(&this->actor) ? 730
+                                                                       : 300);
+    }
```

Comment:
I assume all this is required to match.

Reply:
Yup unfortunately. Very messy.

Reply:
I pulled the ternary out into a separate variable, but it seemingly has to be a ternary at least. Taking it out of the call hopefuly helps a bit.

Reply:
Yeah, that is a significant improvement. You can call it `radius` to match the function arg, too.